stark/src/merkle_tree/commitment.rs

```diff
@@ -0,0 +1,62 @@
+use super::{Error, Hash, Index, Result};
+use itertools::Itertools;
+use std::prelude::v1::*;
+
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Commitment {
+    depth: usize,
+    hash:  Hash,
+}
+
+impl Commitment {
+    pub fn from_depth_hash(depth: usize, hash: &Hash) -> Result<Self> {
```

Comment:
I found this useful discussion about whether to panic or to return a result here: https://www.reddit.com/r/rust/comments/9x17hn/when_should_a_library_panic_vs_return_result/